README.md

```diff
@@ -82,6 +82,92 @@ end
 When pointing the `Vagrantfile` at a manifest instead of directly at a box you retain traditional features such as
 versioning and multiple providers.
 
+Providers
+-------------
+
+There are three AWS credential providers that are used:
+
+* Static credentials
+* Shared credentials
+* Environmental credentials
+
+### Static credentials configuration
+
+Static credentials are used when **both** of the configuration variables `access_key_id` and
+`secret_key` are defined:
+
+```json
+{
+  "post-processors": [
+    [
+      {
+        "type":     "vagrant-s3",
+        ...
+        "access_key_id": "AKIAIOSFODNN7EXAMPLE",
+        "secret_key": "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
+        "session_token": "{{ user `session_token` }}",
+        ...
+      }
+    ]
+  ]
+}
+```
+
+It isn't necessary to define the `session_token` variable, even when using the STS token service for
+authenticating. By default the session token will be generated and cached during the initial request.
+
+The only reason to define `session_token` is when a token has already been generated on behalf of the
+build process via external means (ie: a corporate SSO end-point). You can define this using the
+environmental variable `$AWS_SESSION_TOKEN` or by passing it in from the command line:
+
+```shell
+$ packer build -var "session_token=$TOKEN" template.json
+```
+
+### Shared credentials configuration
+
+Shared credentials are used when **either** of the configuration variables are defined:
+
+```json
+{
+  "post-processors": [
+    [
+      {
+        "type":     "vagrant-s3",
+        ...
+        "credentials": "~/.aws/credentials",
+        "profile": "vagrant-s3-creds",
+      }
+    ]
+  ]
+}
+```
+
+The above configuration will use the `[vagrant-s3-creds]` profile from the credentials file located at
+`~/.aws/credentials`:
+
+```
+[vagrant-s3-creds]
```

Comment:
This was included so there can be zero confusion as to the credentials file location but more importantly how it should be correctly formatted.